app/upgrades/calypso/handler.go

```diff
@@ -0,0 +1,73 @@
+package calypso
+
+import (
+	"fmt"
+
+	gravitytypes "github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	distrkeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	mintkeeper "github.com/cosmos/cosmos-sdk/x/mint/keeper"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	bech32ibckeeper "github.com/osmosis-labs/bech32-ibc/x/bech32ibc/keeper"
+	"github.com/umee-network/umee/x/leverage"
+	leveragekeeper "github.com/umee-network/umee/x/leverage/keeper"
+	leveragetypes "github.com/umee-network/umee/x/leverage/types"
+	"github.com/umee-network/umee/x/oracle"
+	oraclekeeper "github.com/umee-network/umee/x/oracle/keeper"
+	oracletypes "github.com/umee-network/umee/x/oracle/types"
+)
+
+func GetV2UpgradeHandler(
+	mm *module.Manager, configurator *module.Configurator, accountKeeper *authkeeper.AccountKeeper,
+	bankKeeper *bankkeeper.BaseKeeper, bech32IbcKeeper *bech32ibckeeper.Keeper, distrKeeper *distrkeeper.Keeper,
+	mintKeeper *mintkeeper.Keeper, stakingKeeper *stakingkeeper.Keeper, leverageKeeper *leveragekeeper.Keeper,
+	oracleKeeper *oraclekeeper.Keeper,
+) func(
+	ctx sdk.Context, plan upgradetypes.Plan, vmap module.VersionMap,
+) (module.VersionMap, error) {
+	fmt.Println("Enter GetV2UpgradeHandler")
+	if mm == nil || configurator == nil || accountKeeper == nil || bankKeeper == nil || bech32IbcKeeper == nil ||
+		distrKeeper == nil || mintKeeper == nil || stakingKeeper == nil {
+		panic("Nil argument to GetV2UpgradeHandler")
+	}
+	return func(ctx sdk.Context, plan upgradetypes.Plan, vmap module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Calypso upgrade: Enter handler")
+		// We previously upgraded via genesis, thus we don't want to run upgrades for all the modules
```

Comment:
```suggestion
```

Reply:
This comment might want to stay - it made what we're doing more clear to me anyway

Reply:
I'm pretty sure this is a comment from [gravity bridge](https://github.com/Gravity-Bridge/Gravity-Bridge/blob/d8563a0411807768f37ba3d48ee87907bb1f6572/module/app/upgrades/v2/handler.go#L35), idk if it applies

Reply:
Interesting - as far as what the code does, the next lines seems to be basically accepting the current versions of each module as up to date unless we single them out.

If the `we previously upgraded via genesis` need to be removed since it's not us, we might at least comment to note how we're ignoring the `vmap` input parameter which contains previous module versions.

We could even leave a TODO if we suspect that something like SDK 0.46 might cause other modules to require upgrades (and thus their `fromVM` versions).

(Overall pardon my flailing about, this is all new AF to me)

Reply:
You are right... We "upgraded" before, but not real upgrade was done and it wasn't done through the genesis. We should be able to remove that part, I'll test the upgrade that way

Reply:
Yeap, removing those bits. I'm also removing the part in which it overwrites the gravity module version (it was just a workaround for what they did right above it).